cmd/git/git.go

```diff
@@ -1026,167 +1028,76 @@ func (j *DSGit) SetParentCommitFlag(richItem map[string]interface{}) (err error)
-			CalculatedAt:              strfmt.DateTime(j.StatsDt),
-			ProgrammingLanguagesStats: j.ModelPLS,
-			RepositoryURL:             j.URL,
-			TotalLinesOfCode:          int64(j.Loc),
-		},
-	}
-	source := data.DataSource.Slug
+func (j *DSGit) GetModelData(ctx *shared.Ctx, docs []interface{}) []insights.Commit {
+	data := make([]insights.Commit, 0)
+	source := GitDataSource
 	for _, iDoc := range docs {
-		var (
-			message *string
-			title   *string
-		)
+		commit := insights.Commit{}
+		commit.Connector = "git"
+		commit.ConnectorVersion = GitBackendVersion
 		doc, _ := iDoc.(map[string]interface{})
-		docUUID, _ := doc["uuid"].(string)
-		commitURL, _ := doc["commit_url"].(string)
-		commitRepoType, _ := doc["commit_repo_type"].(string)
-		sha, _ := doc["hash"].(string)
-		hashShort, _ := doc["hash_short"].(string)
-		sMessage, _ := doc["message"].(string)
-		sTitle, _ := doc["title"].(string)
-		repoShortURL, _ := doc["repo_short_name"].(string)
-		docCommit, _ := doc["doc_commit"].(bool)
-		emptyCommit, _ := doc["empty_commit"].(bool)
-		locAdded, _ := doc["lines_added"].(int64)
-		locRemoved, _ := doc["lines_removed"].(int64)
-		locChanged, _ := doc["lines_changed"].(int64)
-		_, squashedCommit := j.OrphanedMap["sha"]
-		if sMessage != "" {
-			message = &sMessage
-		}
-		if sTitle != "" {
-			title = &sTitle
-		}
-		parents, _ := doc["parents"].([]string)
-		authoredInTz, _ := doc["author_date"].(time.Time)
+		commit.URL, _ = doc["commit_url"].(string)
+		commit.SHA, _ = doc["hash"].(string)
+		commit.ShortHash, _ = doc["hash_short"].(string)
+		commit.Source = doc["commit_repo_type"].(string)
+		commit.Message, _ = doc["message"].(string)
+		_, commit.MergeCommit = j.OrphanedMap["sha"]
+		commit.ParentSHAs, _ = doc["parents"].([]string)
+		commit.AuthoredTimestamp, _ = doc["author_date"].(time.Time)
 		authoredDt, _ := doc["utc_author"].(time.Time)
-		authoredTz, _ := doc["tz"].(float64)
-		committedInTz, _ := doc["commit_date"].(time.Time)
-		committedDt, _ := doc["utc_commit"].(time.Time)
-		committedTz, _ := doc["commit_tz"].(float64)
+		commit.CommittedTimestamp, _ = doc["commit_date"].(time.Time)
 		createdOn := authoredDt
-		commitRoles := []*models.CommitRole{}
+		commit.Orphaned = doc["orphaned"].(bool)
+		commit.SyncTimestamp = time.Now()
+		commitRoles := []insights.Contributor{}
 		identsAry, okIdents := doc["idents"].([][3]string)
 		identTypesAry, okIdentTypes := doc["ident_types"].([]string)
 		if okIdents && okIdentTypes {
 			for i := range identTypesAry {
-				var (
-					dt     time.Time
-					dtInTz time.Time
-					tz     float64
-				)
+				commitRole := insights.Contributor{}
 				ident := identsAry[i]
 				identType := identTypesAry[i]
-				if identType == "committer" {
-					dt = committedDt
-					dtInTz = committedInTz
-					tz = committedTz
-				} else {
-					dt = authoredDt
-					dtInTz = authoredInTz
-					tz = authoredTz
-				}
+				commitRole.Role = identType
+				commitRole.Weight = 1.0
```

Comment:
Do we have any "default" weights other than 1.0? I mean - everything is 1 by default or maybe some roles have 2 or 0.5 for example?

Reply:
The default weight previously was 1, so I just hardcoded it, we can discuss on role to weightage mapping, I will update the same in the code.

Reply:
We can leave 1, that's fine. Eventually we can have something like weights per role, but this would also need to be hardcoded.